datasets/red_caps/README.md

```diff
@@ -68,6 +69,41 @@ fine-grained descriptions (northern cardinal, taj mahal). Subreddit names provid
 labels (r/shiba) even when captions may not (mlem!), and sometimes may group many visually
 unrelated images through a common semantic meaning (r/perfectfit).
 
+### Dataset Preprocessing
+
+This dataset doesn't download the images locally by default. Instead, it exposes URLs to the images. To fetch the images, use the following code:
+
+```python
+from datasets import load_dataset
+from datasets.utils.file_utils import get_datasets_user_agent
+
+def fetch_images(batch, timeout):
+    import PIL.Image
+    import requests
+
+    images = []
+    for image_url in batch["image_url"]:
+        try:
+            image = PIL.Image.open(
+                requests.get(
+                    image_url,
+                    stream=True,
+                    headers={"user-agent": get_datasets_user_agent()},
```

Comment:
not sure the user agent is necessary but ok :)

Reply:
This fixes the "429: Too many requests" HTTP error that I'd get when trying to download the images on Colab (I can't reproduce this issue locally). Also, img2dataset, a popular tool for building image datasets from image urls, specifies `user-agent` in its [downloader](https://github.com/rom1504/img2dataset/blob/83a757976f4e0ad719068029b2fd41d5a409d13a/img2dataset/downloader.py#L27).